src/cluster.c

```diff
@@ -5925,44 +6118,126 @@ NULL
+             * simply replicates asynchronously the command without the
+             * "REPLICAONLY" marker to the replicas, if there exist any.
+             * The caller is expected to wait for this asynchronous
+             * replication to complete using the "WAIT" command.
+             *
+             * With the help of this command, we finalize the slots
+             * on the replicas before the primary in the following
+             * sequence, where A is the source primary and B is the target
+             * primary:
+             *
+             * 1. Client C issues SETSLOT n NODE B REPLICAONLY against
+             *    node B
+             * 2. Node B replicates SETSLOT n NODE B to all of its replicas,
+             *    such as B', B'', etc
+             * 3. Client C then issues WAIT <num_replicas> <timeout> for
+             *    a number of B's replicas of C's choice to complete the
+             *    finalization
+             * 4. On successful WAIT completion, Client C executes SETSLOT
+             *    n NODE B against node B but without the "REPLICAONLY"
+             *    marker this time, which completes the slot finalization
+             *    on node B
+             *
+             * The following steps can happen in parallel:
+             * a. Client C issues SETSLOT n NODE B against node A
+             * b. Node B gossips its new slot ownership to the cluster,
+             *    including A, A', etc */
+
+            n = clusterLookupNode(c->argv[4]->ptr, sdslen(c->argv[4]->ptr));
+
+            if (!n) {
+                addReplyErrorFormat(c,"Unknown node %s", (char*)c->argv[4]->ptr);
+                return;
+            }
+            if (nodeIsSlave(n)) {
+                addReplyError(c,"Target node is not a master");
+                return;
             }
+            /* If this hash slot was served by 'myself' before to switch
+             * make sure there are no longer local keys for this hash slot. */
+            if (server.cluster->slots[slot] == myself && n != myself) {
+                if (countKeysInSlot(slot) != 0) {
+                    addReplyErrorFormat(c,
+                        "Can't assign hashslot %d to a different node "
+                        "while I still hold keys for this hash slot.", slot);
+                    return;
+                }
+            }
+            if (server.cluster->importing_slots_from[slot] == NULL) {
+                addReplyError(c,"Slot is not open for importing");
+                return;
+            }
+            if (myself->numslaves == 0) {
+                addReplyError(c,"Target node has no replicas");
+                return;
+            }
+
+            /* Remove the last "REPLICAONLY" token so the command
+             * can be applied as the real "SETSLOT" command on the
+             * replicas. */
+            serverAssert(c->argc == 6);
+            c->argc--;
+            decrRefCount(c->argv[5]);
```

Comment:
```suggestion
            rewriteClientCommandVector(c, 5, c->argv[0], c->arv[1], c->argv[2], c->argv[3], c->argv[4]);
```
Kind of surprised there isn't a better API for this, but would like to go through one of the defined rewrite mechanisms.